packages/api/src/router/widgets/indexer-manager.ts

```diff
@@ -0,0 +1,61 @@
+import { TRPCError } from "@trpc/server";
+
+import { ProwlarrIntegration } from "@homarr/integrations";
+import { logger } from "@homarr/log";
+
+import { createManyIntegrationMiddleware } from "../../middlewares/integration";
+import { createTRPCRouter, publicProcedure } from "../../trpc";
+
+export const indexerManagerRouter = createTRPCRouter({
+  getIndexersStatus: publicProcedure
+    .unstable_concat(createManyIntegrationMiddleware("query", "prowlarr"))
+    .query(async ({ ctx }) => {
+      const results = await Promise.all(
+        ctx.integrations.map(async (integration) => {
+          let client;
+          switch (integration.kind) {
+            case "prowlarr":
+              client = new ProwlarrIntegration(integration);
+              break;
+          }
```

Comment:
As already described [here](https://github.com/homarr-labs/homarr/pull/1053#discussion_r1739918181) you can use the existing function that will be typesafe and not show a deepsource issue 😄